### PR TITLE
Synaptic manager app graph

### PR DESCRIPTION
**Pull request by alan-stokes**
removes app graph from dsg

tested by: https://github.com/SpiNNakerManchester/sPyNNaker8/pull/498
fixes: https://github.com/SpiNNakerManchester/sPyNNaker/issues/926

### CODE REVIEW

File: spynnaker/pyNN/models/neuron/synaptic_manager.py [hunk 1]
```diff
@@ -325,8 +325,9 @@ def get_sdram_usage_in_bytes(
 
         :param ~pacman.model.graphs.common.Slice post_vertex_slice:
             The slice of atoms to get the size of
-        :param ~pacman.model.graphs.application.ApplicationGraph \
-                application_graph: The application graph
+        :param application_graph: The application graph
+        :type application_graph: \
+            ~pacman.model.graphs.application.ApplicationGraph
```
Comment (rowleya) [inline spynnaker/pyNN/models/neuron/synaptic_manager.py:330]: Seems pointless...

File: spynnaker/pyNN/models/neuron/synaptic_matrices.py [hunk 2]
```diff
@@ -152,50 +153,56 @@ def __app_matrix(self, app_edge, synapse_info):
...
         """
         gen_on_machine = False
         size = 0
-        for app_edge in app_edges:
+        for edge in edges:
+            if isinstance(edge, MachineEdge):
+                app_edge = edge.app_edge
+            else:
+                app_edge = edge
+
```
Comment (rowleya) [inline spynnaker/pyNN/models/neuron/synaptic_matrices.py:205]: Like the above, I can see this going wrong...

**Review by rowleya — CHANGES_REQUESTED** [review]
Couple of comments made previously (sorry forgot to add to review).  The main issue is that before the code goes through the app vertices and adds the size of each app matrix.  Now it goes through the machine vertices and adds the size of each app matrix.  I hope you can see the difference and how this might lead to overly large SDRAM estimates!  P.S. I don't think we have a test that checks the size of the matrices; maybe we should.

File: spynnaker/pyNN/models/neuron/synaptic_manager.py (hunk 1 above)
Comment (alan-stokes) [reply spynnaker/pyNN/models/neuron/synaptic_manager.py:330]: pointless? when you take it as a arg...... confused

File: spynnaker/pyNN/models/neuron/synaptic_matrices.py [hunk 3]
```diff
@@ -152,50 +153,56 @@ def __app_matrix(self, app_edge, synapse_info):
...
-                for synapse_info in in_edge.synapse_information:
-                    matrix = self.__app_matrix(in_edge, synapse_info)
+        for in_edge in edges:
+            if isinstance(in_edge, MachineEdge):
+                app_edge = in_edge.app_edge
+            else:
+                app_edge = in_edge
+            if isinstance(app_edge, ProjectionApplicationEdge):
+                for synapse_info in app_edge.synapse_information:
+                    matrix = self.__app_matrix(app_edge, synapse_info)
```
[the remark this answers is not in the document]
Comment (alan-stokes) [reply spynnaker/pyNN/models/neuron/synaptic_matrices.py:174]: yea, i know. thats what i learnt late yesterday

File: spynnaker/pyNN/models/neuron/synaptic_matrices.py (hunk 2 above)
Comment (alan-stokes) [reply spynnaker/pyNN/models/neuron/synaptic_matrices.py:205]: i know!

**Comment by alan-stokes** [comment]
we do. funnily enough. coz those tests run out of sdram for some reason.

File: spynnaker/pyNN/models/neuron/synaptic_manager.py (hunk 1 above)
Comment (rowleya) [reply spynnaker/pyNN/models/neuron/synaptic_manager.py:330]: Sorry, I meant the change seems pointless. I think Donal did quite a bit of work to merge the type and the param, but this separates them again.

Comment (alan-stokes) [reply spynnaker/pyNN/models/neuron/synaptic_manager.py:330]: actually. its coz pycharm dislikes the variable name on the next line. it declaires it as not doced. whereas this way pycharm is happy. and its just as good for the docs

**Pull request closed without merging**